Page-level visibility time for embed performance reporting

dvoytenko opened the pull request:
Classical performance reporting against the `navigationStart` time has issues in AMP since AMP pauses all processing while document is invisible. I propose here to also report first/last visibility times.

@tdrl Please guide me what other steps are needed to proceed with these three new metrics?

tdrl commented:
LGTM.  We probably want to add the `{first,last}VisibleTime`s to [ads/google/a4a/google-data-reporter.js#googleLifecycleReporterFactory](https://github.com/ampproject/amphtml/blob/master/ads/google/a4a/google-data-reporter.js#L149).  Not sure what the best parameter to use for that would be -- probably a `met.AD_SLOT_NAMESPACE` of some form.  Maybe Beng or Warren can comment?

dvoytenko commented:
> One small suggestion: Would it make more sense to call it mostRecentVisibleTime instead of lastVisibleTime? When I first read it, I took lastVisible to mean something like "last time before the page is destroyed", rather than "most recent time that page became visible".

This seems like a good suggestion and I was going to do it. But I found bunch of similarly named properties with that exact meaning named "last visible time". So, with that, I'd rather keep it as is. But thanks a lot for suggestion.

dvoytenko commented:
@tdrl Two changes since last review:

1. I pass `met.a4a` query parameters in the reporter.
2. I also signal `preAdThrottle`. I'll be happy to rename if desired. But I wanted to have a directly comparable (and hopefully same name) signal as in 3p that directly pings the `layoutCallback` start.

PTAL.
